crates/bindings-csharp/Runtime/Internal/IIndex.cs

```diff
@@ -0,0 +1,106 @@
+﻿namespace SpacetimeDB.Internal;
+
+using System.Linq;
+
+public interface IIndex<View> where View : IIndex<View> {
+    static abstract string IndexName { get; }
+
+    internal static readonly Lazy<FFI.IndexId> indexId_ =
+        new(() =>
+        {
+            var name_bytes = System.Text.Encoding.UTF8.GetBytes(View.IndexName);
+            FFI.index_id_from_name(name_bytes, (uint)name_bytes.Length, out var out_);
+            return out_;
+        });
+
+    internal static FFI.IndexId indexId => indexId_.Value;
+}
+
+public interface IUniqueIndex<View, Row, T, RW> : IIndex<View>
+    where View : IUniqueIndex<View, Row, T, RW>
+    where Row : ITable<Row>, new()
+    where RW : BSATN.IReadWrite<T>
+{
+    private static byte[] Params(T key, RW bsatn, out uint idx, out uint end) {
+        using var s = new MemoryStream();
+        using var w = new BinaryWriter(s);
+        w.Write(0); bsatn.Write(w, key); // inclusive rstart
```

Comment:
These are kinda dangerous, I don't like relying on inference to ensure that `Write(0)` will result in the correct integer type.

Can we use a `const` with explicit type? That would also help readability to clarify what this zero represents.